packages/twenty-front/src/modules/serverless-functions/hooks/useTestServerlessFunction.ts

```diff
@@ -0,0 +1,63 @@
+import { SnackBarVariant } from '@/ui/feedback/snack-bar-manager/components/SnackBar';
+import { useExecuteOneServerlessFunction } from '@/settings/serverless-functions/hooks/useExecuteOneServerlessFunction';
+import { useRecoilState } from 'recoil';
+import { serverlessFunctionTestDataFamilyState } from '@/workflow/states/serverlessFunctionTestDataFamilyState';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { isDefined } from 'twenty-ui';
+
+export const useTestServerlessFunction = (
+  serverlessFunctionId: string,
+  callback?: (testResult: object) => void,
+) => {
+  const { enqueueSnackBar } = useSnackBar();
+  const { executeOneServerlessFunction } = useExecuteOneServerlessFunction();
+  const [serverlessFunctionTestData, setServerlessFunctionTestData] =
+    useRecoilState(serverlessFunctionTestDataFamilyState(serverlessFunctionId));
+
+  const testServerlessFunction = async () => {
+    try {
+      const result = await executeOneServerlessFunction({
+        id: serverlessFunctionId,
+        payload: serverlessFunctionTestData.input,
+        version: 'draft',
+      });
+
+      if (isDefined(result?.data?.executeOneServerlessFunction?.data)) {
+        callback?.(result?.data?.executeOneServerlessFunction?.data);
+      }
+
+      setServerlessFunctionTestData((prev) => ({
+        ...prev,
+        language: 'json',
+        height: 300,
+        output: {
+          data: result?.data?.executeOneServerlessFunction?.data
+            ? JSON.stringify(
+                result?.data?.executeOneServerlessFunction?.data,
+                null,
+                4,
+              )
+            : undefined,
+          duration: result?.data?.executeOneServerlessFunction?.duration,
+          status: result?.data?.executeOneServerlessFunction?.status,
+          error: result?.data?.executeOneServerlessFunction?.error
+            ? JSON.stringify(
+                result?.data?.executeOneServerlessFunction?.error,
+                null,
+                4,
+              )
+            : undefined,
+        },
+      }));
+    } catch (err) {
+      enqueueSnackBar(
+        (err as Error)?.message || 'An error occurred while executing function',
+        {
+          variant: SnackBarVariant.Error,
+        },
+      );
+    }
+  };
```

Comment:
Later, we'll have to prevent users from clicking these buttons multiple times while an operation is pending. We'll have to do it in many places to avoid unwanted behaviors.

Reply:
Updating or creating a serverless function is almost instant now. The serverless lambda build and deployment which can take several seconds, is now handled asynchronously with a job
The only weird bahavior that can remain is that when you test your function just after updating it, the result my be the result of the previous version during the deployment. It will be address with a while loop that check if function is properly deployed